### ISSUE TITLE
Single vs Double Quotes Issue Diff

### ISSUE THREAD

**Issue by ryanmeekins** [tconbeer/sqlfmt#400]
**Describe the bug**
When running `sqlfmt`, sometimes it doesn't format single quotes (`'`) to double quotes (`"`). However, when running `sqlfmt . --diff` in a GitHub action, it always seems to detect that it needs to change single quotes to double quotes.

**To Reproduce**
It's not easy to reproduce as I think it's a cache issue. But I have a file `/snapshots/agreement_contacts_snapshot.sql` as:
```
{% snapshot agreement_contacts_snapshot %}

{{
    config(
      unique_key="proj_num || '-' || primary_bp_no",
      strategy='timestamp',
      updated_at='last_updated_at',
    )
}}

select *
from {{ ref('agreement_contacts') }}

{% endsnapshot %}
```
When running `sqlfmt` , it wouldn't reformat the file for a while. I was using `sqlfmt .`, `sqlfmt snapshots/`, and even `sqlfmt snapshots/ -k` to try and reset the catch. However, running `sqlfmt . --diff` does show this file has changes. And then it eventually does reformat the file, possibly the next time after running the `-k` argument.

**Expected behavior**
I would expect it to format this file when it's needed, instead of having to try lots of times. I'm not sure why some files seem to have this issue. We are using `pre-commit` with `sqlfmt`, so we'd expect it to format files as we push commits. However, sometimes it doesn't and then our GitHub action that runs `sqlfmt . --diff` fails because of this formatting issue. This has happened quite a few times in the past month.

**Actual behavior**
![image](https://user-images.githubusercontent.com/36648892/231231154-515a97d7-e983-4699-bfc1-aa4b5a696299.png)

**Additional context**
sqlfmt, version 0.17.0

**Comment by tconbeer** [comment]
Black is doing the formatting here, since we're inside jinja. Are you sure you have black installed or installed sqlfmt with the jinjafmt extra?

**Comment by ryanmeekins** [comment]
Yes, black is installed. I also have it in pre-commit. We do have it listed separately in our `requirements.txt` as well as `shandy-sqlftm[jinjafmt]`
```
$ black --version
black, 22.12.0 (compiled: yes)
Python (CPython) 3.10.10
```
Could it be a cache issue with `sqlfmt` not handing the text off to `black` for processing?

**Comment by ryanmeekins** [comment]
Also, this file was eventually formatted correctly, just not sure why it wasn't earlier during the `pre-commit` run. Here's our `.pre-commit-config.yaml`:
```
repos:
  - repo: https://github.com/tconbeer/sqlfmt
    rev: v0.17.0
    hooks:
      - id: sqlfmt
        language_version: python
  - repo: https://github.com/ambv/black
    rev: 22.12.0
    hooks:
      - id: black
        language_version: python3.7
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
      - id: trailing-whitespace
      - id: end-of-file-fixer
      - id: check-yaml
 ```

**Comment by tconbeer** [comment]
Re-reading this more carefully now -- pre-commit passes a list of the changed files to sqlfmt, so when running sqlfmt in pre-commit, it doesn't run against your whole repo, only the files with changes. When you run `sqlfmt . --diff` in a GHA, that runs against the whole repo. Could that be causing your problem?

**Comment by tconbeer** [comment]
One other thing to test -- I don't think pre-commit installs jinjafmt unless you tell it to. Each pre-commit hook runs in its own virtual environment, so i don't think it helps that you have black installed locally or as another hook. Can you try this for your pre-commit config? If that works, I'll update the docs:

```yml
repos:
  - repo: https://github.com/tconbeer/sqlfmt
    rev: v0.17.0
    additional_dependencies: ['shandy-sqlfmt[jinjafmt]']
    hooks:
      - id: sqlfmt
        language_version: python
```

You could also try:


```yml
repos:
  - repo: https://github.com/tconbeer/sqlfmt
    rev: v0.17.0
    additional_dependencies: ['black']
    hooks:
      - id: sqlfmt
        language_version: python
```

**Comment by ryanmeekins** [comment]
> Re-reading this more carefully now -- pre-commit passes a list of the changed files to sqlfmt, so when running sqlfmt in pre-commit, it doesn't run against your whole repo, only the files with changes. When you run `sqlfmt . --diff` in a GHA, that runs against the whole repo. Could that be causing your problem?

I don't think so because as you can see in the screenshot, I did run `sqlfmt` locally and I would think that it should have updated the file (but didn't at first). It eventually did but not the first time as expected. And these files should be formatted as we're making changes and pushing commits. We've had `pre-commit` set up on our project for months now and I just created that example snapshot this morning.

**Comment by ryanmeekins** [comment]
Also, I'm looking into changes the pre-commit config now.

**Comment by ryanmeekins** [comment]
I just updated my pre-commit config and tested it on the sample file, and it worked! I'm still not sure why running `sqlfmt` locally didn't work, but I'm hoping that this will at least catch issues before the GHA. I appreciate you helping out with this so quickly. Also, the `additional_dependencies` needs to be under the `hooks`, like this:

```yaml
repos:
  - repo: https://github.com/tconbeer/sqlfmt
    rev: v0.17.0
    hooks:
      - id: sqlfmt
        language_version: python
        additional_dependencies: ['shandy-sqlfmt[jinjafmt]']
```

**Comment by tconbeer** [comment]
[docs](https://docs.sqlfmt.com/integrations/pre-commit) updated. Thanks, @ryanmeekins

**Comment by ivan-toriya-precis** [comment]
omg, that's finally solved the same problem #397 that I had. Thank you both!
